tests/System.Buffers.ReaderWriter.Tests/Reader_SkipDelimiter.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers.Reader;
+using System.Globalization;
+using System.Text;
+using Xunit;
+
+namespace System.Buffers.Tests
+{
+    public class Reader_SkipDelimiter
+    {
+        [Fact]
+        public void TryReadTo_SkipDelimiter()
+        {
+            byte[] expected = Encoding.UTF8.GetBytes("This is our \\\"understanding\\\"");
+            ReadOnlySequence<byte> bytes = BufferFactory.CreateUtf8("This is our \\\"understanding\\\"\" you see.");
+            BufferReader<byte> reader = new BufferReader<byte>(bytes);
+            Assert.True(reader.TryReadTo(out ReadOnlySpan<byte> span, (byte)'"', (byte)'\\', advancePastDelimiter: true));
+            Assert.Equal(expected, span.ToArray());
+            Assert.True(reader.IsNext((byte)' '));
+            Assert.Equal(30, reader.Consumed);
+
+            reader = new BufferReader<byte>(bytes);
+            Assert.True(reader.TryReadTo(out span, (byte)'"', (byte)'\\', advancePastDelimiter: false));
```

Comment:
this API will probably be used for reading strings mainly. It might be better to take UnicodeScalar, especially if we have support from the language for crating scalar literals. Casing chars to bytes does not scale beyond ascii and the syntax is ugly. @GrabYourPitchforks

Reply:
Performance tanks for non ASCII and most of the time (90% plus?) parsing involves ASCII delimiters. We can always create full Unicode specific ones that might even call to this in some cases.